### ISSUE TITLE
Show phenotype annotation from has_part rels

### ISSUE THREAD

**Issue by kimrutherford** [pombase/website#89]
On the term pages and for counts, include has_part relationships but handle them specially.

From @mah11:

There should be lots of has_part links between FYPO terms, and they should be followed the same way as it works in Canto. Example:

id: FYPO:0000223
name: elongated multiseptate vegetative cell
relationship: has_part FYPO:0000061 ! multinucleate vegetative cell
relationship: has_part FYPO:0001122 ! elongated vegetative cell

The counts for FYPO:0000061 and FYPO:0001122 should include things annotated to FYPO:0000223 (and search results should do the same, when you get there).

**Comment by mah11** [comment]
remember that this is just for FYPO; as far as I know, we still don't want to include has_part for generating GO annotation counts

**Comment by kimrutherford** [comment]
> remember that this is just for FYPO

OK, thanks.

**Comment by kimrutherford** [comment]
I've had a go at this.  It's live in pombase v2 but I haven't done a thorough check of the counts yet.  Might need to fix #136 first.

**Comment by ValWood** [comment]
Yep, I can't check it, because I won't know if the counts are correct by looking because I will only know if they are right for single gene phenotypes...
All we will know is they will be the same, or higher than the numbers on the current site...

**Comment by kimrutherford** [comment]
I've had a look at a few FYPO terms that have a has_part and things seem OK.  I haven't double checked any counts but it looks like the right terms and annotations appear on pages like this:
  http://pombase2.bioinformatics.nz/term/FYPO%3A0000133

**Comment by ValWood** [comment]
For has_part, this is definitely the wrong way around for inference.

So 

	elongated multinucleate vegetative cell
has part FYPO:0003341multinucleate
has_part FYPO:0000287	abnormal subcellular component

on the page.

Something is odd here
i) we would not expect elongated multinucleate vegetative cell
to inherit annotations from
 FYPO:0000287	abnormal subcellular component 

ii) the second odd thing is that I wouldn't expect any direct annotations to 
FYPO:0000287	abnormal subcellular component
so I think this is somehow just populating and displaying all of the "has_parts"

This is tru, but it  isn't necessary in the context of "what are the annotations to elongated multinucleate vegetative cell"

What we really want to display is each annotation, only once, and the term it is derived from....

**Comment by mah11** [comment]
Yes, this does look like it's going backwards relative to what we want. With has_part, we want to use it for phenotype annotation propagation in the direction whole-implies-part. If FYPO has A has_part B, we should see genes/genotypes annotated to A on the page for B, and we should not see things annotated to directly to B on the page for A. The page for A also shouldn't show things annotated to B that B inherits via is_a or part_of paths.

So in the FYPO:0000133 example, the ontology has

    FYPO:0000133
    has_part FYPO:0000061 ! multinucleate vegetative cell
    has_part FYPO:0001122 ! elongated vegetative cell

and (among others)

    FYPO:0001253
    is_a: FYPO:0000133 ! elongated multinucleate vegetative cell

The page for FYPO:0000133 should show things annotated directly to FYPO:0000133 or FYPO:0001253. It should not show everything that is annotated to FYPO:0000061 or FYPO:0001122.

The pages for FYPO:0000061 and FYPO:0001122 should include everything annotated to FYPO:0000133 (and therefore also FYPO:0001253).

(this might be relevant, on swapping terms - https://github.com/pombase/canto/issues/590#issuecomment-119428858)

**Comment by ValWood** [comment]
Thank you for making things clearer. I'm so glad you have not got a TBI...

**Comment by kimrutherford** [comment]
> Yes, this does look like it's going backwards relative to what we want.

Drat!  I thought I had it right.  I'll read through all the comments to make sure I'm following, then I'll try to fix it.

**Comment by kimrutherford** [comment]
> If FYPO has A has_part B, we should see genes/genotypes annotated to A on the page for B

So the same as for is_a and the other relations?  That's why it's backwards on the pages.  I was doing the swapping as in the comment you linked to: https://github.com/pombase/canto/issues/590#issuecomment-119428858

Currently if A is_a B the annotations for A appear on the B page.  For FYPO, if A has_part B the annotations for B appear on the A page.  I'll make has_part act like is_a.

**Comment by mah11** [comment]
OK. I remember both (all? or maybe V&A had the sense to stay out of it) going back & forth getting confused over whether to swap or not. Certainly now do whichever makes the pages look right!

**Comment by kimrutherford** [comment]
I've changed the handling of has_part so now Midori's FYPO:0000133 example works as described.  As usual has_part is making my head hurt.

The reason I got this wrong is that I copied what I did in Canto.  But I should have realised that since the web code is uses the ontology closure in Chado that is generated by Owltools, all the inference and special handling of relations like has_part is already done.  I was undoing some of the work done by Owltools and so the display was wrong.

Could you have a spot check to see if things look right now?

**Comment by mah11** [comment]
Yes, the included terms look sensible now. Thanks!

> As usual has_part is making my head hurt.

You are not alone.

**Comment by ValWood** [comment]
Its the hardest part.....plain sailing now ;)

**Comment by ValWood** [comment]
I'm still a little confused.
There are no has_parts here now (that might be correct)
http://pombase2.bioinformatics.nz/term/FYPO%3A0000133

**Comment by mah11** [comment]
That's correct - there are no "has_part FYPO:0000133" links in FYPO. Try FYPO:0000061 or FYPO:0001122.

**Comment by ValWood** [comment]
looks good...

**Comment by kimrutherford** [comment]
I'll close this now since it seems OK (cross fingers).  Please open new issues for bugs.
